### PR TITLE
Based on the Kleisli, re-implement all of the API

### PR DESCRIPTION
**Pull request by tkrs**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,6 +19,75 @@ libraryDependencies ++= Seq(
...
+
+  try {
+    val action = for {
+      _ <- put(settings, u1)
+      get <- get[User](settings)
+      del <- delete(settings)
+      _ <- puts(settings, Map(u1.name -> u1, u2.name -> u2, u3.name -> u3))
+      all <- all[User](settings)
+      dels <- deletes(settings, Seq(u1.name, u2.name, u3.name))
+    } yield get :: del :: all :: dels :: HNil
```
Comment (yanana) [inline README.md:58]: In the example, this action doesn't need to yield `HList`.

File: core/src/main/scala/aerospiker/command/ReadHeader.scala [hunk 2]
```diff
@@ -17,12 +20,12 @@ class AsyncReadHeader[T](cluster: AsyncCluster, policy: Policy, listener: Option
...
   def getPolicy: Policy = policy
 
-  def writeBuffer: Unit = setReadHeader(policy, key)
+  def writeBuffer(): Unit = setReadHeader(policy, key)
 
   def getNode: AsyncNode =
     cluster.getReadNode(partition, policy.replica).asInstanceOf[AsyncNode]
 
-  final def parseResult(byteBuffer: ByteBuffer): Unit = {
+  def parseResult(byteBuffer: ByteBuffer): Unit = {
```
Comment (yanana) [inline core/src/main/scala/aerospiker/command/ReadHeader.scala:28]: We should move byte operations to scodec someday :-)

File: core/src/main/scala/aerospiker/Functions.scala [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package aerospiker
+
+import cats.data.ReaderT
+
+trait Functions {
+  type C
+  def withC[F[_], U](f: C => F[U]): Action[F, C, U] = ReaderT.function[F, C, U](f)
```
Comment (yanana) [inline core/src/main/scala/aerospiker/Functions.scala]: There might be some more appropriate name...

File: core/src/main/scala/aerospiker/command/ReadHeader.scala (hunk 2 above)
Comment (tkrs) [inline core/src/main/scala/aerospiker/command/ReadHeader.scala:28]: yes:sweat_smile:

File: core/src/main/scala/aerospiker/Functions.scala (hunk 3 above)
Comment (tkrs) [inline core/src/main/scala/aerospiker/Functions.scala]: `withClient` instead?

File: README.md (hunk 1 above)
Comment (tkrs) [inline README.md:58]: Exactly!

**Comment by yanana** [comment]
LGTM

**Pull request merged**